src/ssh/ssh_client.cpp

```diff
@@ -154,5 +154,17 @@ int mp::SSHClient::exec_string(const std::string& cmd_line)
 
     handle_ssh_events();
 
-    return ssh_channel_get_exit_status(channel.get());
+    uint32_t exit_status = static_cast<uint32_t>(-1);
+    char* exit_signal_status = nullptr;
+    int result = ssh_channel_get_exit_state(channel.get(), &exit_status, &exit_signal_status);
+    
+    switch (result) {
+        case SSH_OK:
+            return exit_status;
+        case SSH_AGAIN:
+            return -1;
+        case SSH_ERROR:
+        default:
+            return -1;
```

Comment:
I think it may be cleaner if you rewrite it using the ternary operator:

```suggestion
    return result == SSH_OK ? exit_status : -1;
```